newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -131,227 +79,200 @@ type Props = {|
-      state.inputValue !== null ? state.inputValue : props.value;
+    const { classes } = this.props;
+
+    const getCurrentInputValue = (): string =>
+      this.state.inputValue !== null ? this.state.inputValue : this.props.value;
+
+    const filterFunction = (options: DataSource, state: Object): DataSource => {
+      const lowercaseInputValue = getCurrentInputValue().toLowerCase();
+      const optionList = options.filter(option => {
+        if (option.type === 'separator') return true;
+        if (!option.text) return true;
+        return option.text.toLowerCase().indexOf(lowercaseInputValue) !== -1;
+      });
 
-    const helperText = props.helperMarkdownText ? (
-      <MarkdownText source={props.helperMarkdownText} />
+      if (
+        !optionList.filter(
+          option => option.type !== 'separator' && option.value
+        ).length
+      )
+        return [];
+
+      // Remove divider(s) if they are at the start or end of array
+      while (
+        optionList[optionList.length - 1] !== undefined &&
+        optionList[optionList.length - 1].type !== undefined
+      )
+        optionList.pop();
+      while (
+        optionList[optionList.length - 1] !== undefined &&
+        optionList[0].type !== undefined
+      )
+        optionList.shift();
+
+      return optionList;
+    };
+
+    const helperText = this.props.helperMarkdownText ? (
+      <MarkdownText source={this.props.helperMarkdownText} />
     ) : null;
 
+    const getOptionLabel = (option: Option): string => {
+      if (option.value) return option.value;
+      return getCurrentInputValue();
+    };
+
+    const handleChange = (event, option: Option | string): void => {
+      if (option !== null && option.type !== 'separator') {
+        if (typeof option === 'string') {
+          this.props.onChange(option);
+        } else {
+          if (option.onClick) option.onClick();
+          else {
+            if (this.props.onChoose) this.props.onChoose(option.value);
+            else this.props.onChange(option.value);
+          }
+        }
+      }
+      if (
+        typeof event.key === 'string' &&
+        event.key === 'Enter' &&
+        this._input.current !== null
+      )
+        this._input.current.blur();
+      //if (this.props.onRequestClose) this.props.onRequestClose();
+    };
+
+    const handleInputChange = (e, value: string, reason: string): void =>
+      this.setState({ inputValue: value });
+
+    const getDefaultStylingProps = (params: Object): Object => {
+      const { InputProps, inputProps, InputLabelProps, ...other } = params;
+      return {
+        ...other,
+        InputProps: {
+          ...InputProps,
+          className: null,
+          endAdornment: null,
+        },
+        inputProps: {
+          ...inputProps,
+          className: null,
+          onKeyDown: (event): void => {
+            // if (event.key === 'Escape' && this.props.onRequestClose)
+            //   this.props.onRequestClose();
```

Comment:
Remove the commented code ;) 
The code living in the codebase must be as simple as possible - dead code is 99% of the time just making things harder to understand. If you want to keep it, keep a separate branch on your local computer or your own git with this code.

Reply:
Im really sorry, should've checked before pushing, my bad.

Reply:
Im really sorry, should've checked before pushing, my bad.

Reply:
No worries :)